images/virtualization-artifact/pkg/controller/vm/internal/lifecycle.go

```diff
@@ -58,6 +58,22 @@ func (h *LifeCycleHandler) Handle(ctx context.Context, s state.VirtualMachineSta
 	current := s.VirtualMachine().Current()
 	changed := s.VirtualMachine().Changed()
 
+	defer func() {
+		if changed == nil {
+			return
+		}
+		if len(changed.Status.Conditions) == 0 {
+			changed.Status.ObservedGeneration = changed.GetGeneration()
+			return
+		}
```

Comment:
Shouldn't we always update ObservedGeneration on every reconcile despite of changes?

Reply:
I think we should update the ObservedGeneration when the ObservedGeneration of all conditions has been updated